api/python/provisioner/salt.py

```diff
@@ -410,6 +425,36 @@ def _get_state_fails(self, ret: Dict):
         return fails
 
 
+@attr.s(auto_attribs=True)
+class SaltCallerClientResult(SaltClientResult):
+    def __attrs_post_init__(self):
+        self.raw = {
+            'local': {
+                'ret': self.raw,
+                # XXX not declared salt internals, no other way to
+                #     detect a failure
+                'retcode': self.client.sminion.functions.pack[
+                    '__context__'].get('retcode', 0)
+            }
+        }
+        super().__attrs_post_init__()
+
+    def _parse_raw_dict(self):
+        ret = self.raw['local']['ret']
+
+        if (
+            self.cmd_args_view['fun'].startswith('state.')
+            and (type(ret) is dict)
```

Comment:
Based on PEP8:
https://www.flake8rules.com/rules/W503.html
The best practice it is place binary operator `and` before line break:
```
if (
self.cmd_args_view['fun'].startswith('state.') and
(type(ret) is dict)
}
```

I also prefer to do the additional indent for the second and following lines with conditions
```
        if (self.cmd_args_view['fun'].startswith('state.') and 
               (type(ret) is dict)):
```
From my point of view, such way is more readable and widely used in other projects. But it is not required by PEP8.

Reply:
Sorry, neither I nor pep8 can agree with you here. Please check the link you provided (it's not a pep8 but how flake8 interprets it which is a different thing and looks outdated) and the [link](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator) which is referred there more carefully:
```
In Python code, it is permissible to break before or after a binary operator, as long as the convention is consistent locally. For new code Knuth's style is suggested.
```
and
```
Donald Knuth explains the traditional rule in his Computers and Typesetting series: "Although formulas within a paragraph always break after binary operations and relations, displayed formulas always break before binary operations"
```
We don't have a good habit in our code for the moment but generally I believe the option of "before operator" is presented much and preffered.

That PR is not intended to make a style clean up for the whole repo and is quite aligned with the current code base. If you want to change something globally please raise a discussion inside a team and then a separate PR might be created.

Reply:
I agree that it is discussable. So my main message here is even bad code guidelines are better when all follow them. About pep8, yes I investigated it before my original post. You can also see that even pep8 mesh different ways:
https://www.python.org/dev/peps/pep-0008/#code-lay-out
```
# No extra indentation.
if (this_is_one_thing and
    that_is_another_thing):
    do_something()

# Add a comment, which will provide some distinction in editors
# supporting syntax highlighting.
if (this_is_one_thing and
    that_is_another_thing):
    # Since both conditions are true, we can frobnicate.
    do_something()

# Add some extra indentation on the conditional continuation line.
if (this_is_one_thing
        and that_is_another_thing):
    do_something()
```
Also need to remember that flake8 was widely used in Seagate to check PEP8 guidelines before codacy. And flake8 treats them as warnings which we in CSM/Provisioner fixed.
But as you mention, the new approach for line breaks appeared as the best practice in the last few years. So as result we need to choose one method and use it always.

Reply:
>Also need to remember that flake8 was widely used in Seagate to check PEP8 guidelines before codacy. And flake8 treats them as warnings which we in CSM/Provisioner fixed.

I had checked the PR using flake8 before requested for a review and I didn't see any warnings, not sure why are you mentioning codacy here.

So to summarize:
- I don't see here any distraction with neither recommended way (pep8) nor existent codebase
- flake8 check is passed
- broad changes are out of the PR's scope